### DISCUSSION TITLE
Compiling from git source fails

### DISCUSSION THREAD

**Discussion by VictorEijkhout** [idaholab/moose#26975] [Q&A Getting Started]
even after pulling all the submodules I get:
```
make[1]: Entering directory '/work/00434/eijkhout/moose/moose-git/test'
/bin/sh: /work/00434/eijkhout/moose/moose-git/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
```

**Comment by dschwen** [comment]
Did you run `./scripts/update_and_rebuild_libmesh.sh`? Are you using (and activating) our conda environment?

**Comment by VictorEijkhout** [comment]
Was that script mentioned in the installation instructions? No, I didn't run it.

And I prefer to leave conda out of it because it messes with my environment.

Also: how do I specify an installation prefix location?

**Reply by GiudGiud** [reply]
Which installation page are you following?
Conda is good at not messing with the existing environment. Unless nearly all the other tools (brew & pip are the worst offenders)

**Comment by VictorEijkhout** [comment]
```
Cloning MOOSE

MOOSE is hosted on [GitHub](https://github.com/idaholab/moose) and should be cloned directly from there using [git](https://git-scm.com/). We recommend creating a directory ~/projects to contain all of your MOOSE related work.

To clone MOOSE, run the following commands in a terminal:

mkdir -p ~/projects
cd ~/projects
git clone https://github.com/idaholab/moose.git
cd moose
git checkout master
Copy

commentnote

The master branch of MOOSE is the stable branch that will only be updated after all tests are passing. This protects you from the day-to-day changes in the MOOSE repository.
Build and Test MOOSE

To build MOOSE run the following commands:

cd ~/projects/moose/test
make -j 6
```

**Reply by GiudGiud** [reply]
what s the link to this page? how did you reach it?
this is assuming the dependencies have already been installed

**Reply by milljm** [reply]
This information is on the Conda installation guide: https://mooseframework.inl.gov/getting_started/installation/conda.html, Under "Cloning MOOSE".

I think the issue involved, is due to a new terminal window being opened? And thus, we need to perform another `conda activate moose`, before proceeding with `make`

**Reply by VictorEijkhout** [reply]
Well, that page is linked as 
```
Installing MOOSE

To begin, click a link corresponding to your operating system or another installation method of choice:

    [Linux and MacOS](https://mooseframework.inl.gov/getting_started/installation/conda.html)
```
so it's not specific to Conda, so I skipped the conda instructions, and when to the clone & build section.

**Comment by VictorEijkhout** [comment]
@dschwen Ok, I'm calling `install_and_rebuild` which ends on 
```
<<< Found PETSc 3.20.3 installation in /work/00434/eijkhout/moose/moose-git/scripts/../petsc ... >>>
checking whether we can compile a trivial PETSc program... no
checking for TAO support via PETSc... no
configure: error: *** PETSc was not found, but --enable-petsc-required was specified.
```

How do I debug that?

**Reply by dschwen** [reply]
You run `./scripts/update_and_rebuild_petsc.sh` first

**Reply by VictorEijkhout** [reply]
Can you point me to full instructions for building from source?

**Reply by dschwen** [reply]
Take a look at https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html

I think the root of the problem is that we don't explicitly cater to people who shun our Conda environment. The implicit assumption on our part is probably that those people don't require as much hand-holding. But they do deserve a clear pointer on our docs landing page.

**Reply by VictorEijkhout** [reply]
Ok, I'll follow those instructions.

Eh, for installation on an HPC cluster I would expect a way to a set an installation prefix. That page assumes I'm installing in my home directory, which is not very HPC'ish, if you ask me. Which you didn't but I'm volunteering.

**Reply by milljm** [reply]
Part of the problem is that we run into so many folks that will literally cut&paste:

```bash
./update_and_rebuild_petsc --prefix=/path/to/desired/location
```
runs into an error, and then post it here about not having access to `/path`. 🤦 

But, we can always assume `~/` is available. For the power-user such as yourself, I suppose I should create a specific help document. One that lists our working `configure --` arguments, and that's about it. I myself would probably enjoy a page like this. Instead of having to `cat moose/scripts/configure_petsc|libmesh|wasp.sh` to see what I need.
